pkg/chunk/storage/factory.go

```diff
@@ -116,6 +124,10 @@ func NewStore(cfg Config, storeCfg chunk.StoreConfig, schemaCfg chunk.SchemaConf
 
 // NewIndexClient makes a new index client of the desired type.
 func NewIndexClient(name string, cfg Config, schemaCfg chunk.SchemaConfig) (chunk.IndexClient, error) {
+	if factory, isOK := customIndexClients[name]; isOK {
```

Comment:
Please add a unit test on this. You will need a `UnregisterIndexClient` function to call it in a `defer`within the test, so that we cleanup the global state before the test exits.